components/carousel/carousel.js

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getClassNamesWithMods, getDataAttributes } from '../_helpers';
+
+import CarouselItem from './carouselItem';
+import CarouselMarkers from './carouselMarkers';
+import CarouselPage from './carouselPage';
+import Swipe from '../../utils/swipe';
+
+/**
+ * Carousel component
+ */
+export default class Carousel extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      images: props.images,
+      isTransitioning: false,
+      previousItem: null,
+      currentItem: 0,
+    };
+
+    this.handleClickNext = this.handleClickNext.bind(this);
+    this.handleClickPrev = this.handleClickPrev.bind(this);
+    this.handleClickGoTo = this.handleClickGoTo.bind(this);
+    this.handleSwipeNext = this.handleSwipeNext.bind(this);
+    this.handleSwipePrev = this.handleSwipePrev.bind(this);
+  }
+
+  componentDidMount() {
+    const swipeHandler = new Swipe(this.carouselTracker);
+    swipeHandler.onLeft(this.handleSwipeNext);
+    swipeHandler.onRight(this.handleSwipePrev);
+  }
+
+  setNextItem(next) {
+    this.setState(state => ({
+      isTransitioning: true,
+      currentItem: next,
+      previousItem: state.currentItem,
+    }));
+  }
+
+  handleClickPrev(e) {
+    e.preventDefault();
+    let next = this.state.currentItem - 1;
+    if (next < 0) {
+      next = this.props.images.length - 1;
+    }
+
+    this.setNextItem(next);
+  }
+
+  handleClickNext(e) {
+    e.preventDefault();
+
+    let next = this.state.currentItem + 1;
+    if (next === this.props.images.length) {
+      next = 0;
+    }
+
+    this.setNextItem(next);
+  }
+
+  handleSwipePrev() {
+    let next = this.state.currentItem - 1;
+    if (next < 0) {
+      return;
```

Comment:
Because when swiping, it'd be _unnatural_ to swipe and throw all the way to the other side. It's in the roadmap to do the trick about adding the last before the first and first after last in order to create that.